instance/models/openedx_appserver.py

```diff
@@ -0,0 +1,248 @@
+    ))
+    protocol = models.CharField(max_length=5, default='http', choices=PROTOCOL_CHOICES)
+
+    # Ansible-specific settings:
+    configuration_source_repo_url = models.URLField(max_length=256, blank=False)
+    configuration_version = models.CharField(max_length=50, blank=False)
+    configuration_extra_settings = models.TextField(blank=True, help_text="YAML config vars that override all others")
+
+    edx_platform_repository_url = models.CharField(max_length=256, blank=False, help_text=(
+        'URL to the edx-platform repository to use. Leave blank for default.'
+    ))
+    edx_platform_commit = models.CharField(max_length=256, blank=False, help_text=(
+        'edx-platform commit hash or branch or tag to use. Leave blank to use the default, '
+        'which is equal to the value of "openedx_release".'
+    ))
+
+    openedx_release = models.CharField(max_length=128, blank=False, help_text=("""
+        Set this to a release tag like "named-release/dogwood" to build a specific release of
+        Open edX. The default is "{default}". This setting becomes the default value for
+        edx_platform_version, forum_version, notifier_version, xqueue_version, and
+        certs_version so it should be a git branch that exists in all of
+        those repositories.
+
+        Note: to build a specific branch of edx-platform, you should just override
+        edx_platform_commit rather than changing this setting.
+
+        Note 2: This value does not affect the default value of configuration_version.
+    """).format(default=settings.DEFAULT_OPENEDX_RELEASE))
+
+    s3_access_key = models.CharField(max_length=50, blank=True)
+    s3_secret_access_key = models.CharField(max_length=50, blank=True)
+    s3_bucket_name = models.CharField(max_length=50, blank=True)
+
+    # Misc settings:
+    use_ephemeral_databases = models.BooleanField()
+    github_admin_organization_name = models.CharField(
+        max_length=200, blank=True, default=settings.DEFAULT_ADMIN_ORGANIZATION,
+        help_text="GitHub organization whose users will be given SSH access to this instance's VMs",
+    )
+
+    def set_field_defaults(self):
+        """
+        Set default values.
+        """
+        if not self.openedx_release:
+            self.openedx_release = settings.DEFAULT_OPENEDX_RELEASE
+        if not self.configuration_source_repo_url:
+            self.configuration_source_repo_url = settings.DEFAULT_CONFIGURATION_REPO_URL
+        if not self.configuration_version:
+            self.configuration_version = settings.DEFAULT_CONFIGURATION_VERSION
+        if not self.edx_platform_repository_url:
+            self.edx_platform_repository_url = 'https://github.com/{}.git'.format(settings.DEFAULT_FORK)
+        if not self.edx_platform_commit:
+            self.edx_platform_commit = self.openedx_release
+        super().set_field_defaults()
+
+
+class OpenEdXAppServer(AppServer, OpenEdXAppConfiguration, AnsibleAppServerMixin, EmailMixin):
```

Comment:
@itsjeyd If only one of the models subclassed `OpenEdXAppConfiguration`, then the other model would not be able to store the data. e.g. If I removed it from AppServer, then how would this code work:

```python
        app_server = self.appserver_set.create(
            name="AppServer {}".format(self.appserver_set.count() + 1),
            # Copy the current value of each setting into the AppServer, preserving it permanently:
            configuration_database_settings=self.get_database_settings(),
            **instance_config
        )
```

The `OpenEdXAppServer` would not have any fields that can hold the `instance_config` data, and it would cause an error.

I *could* add a generic `JSONField` to store the copy of all the config values, but I think that's not as nice.
We could also not store them and just store the resulting 'configuration_settings' YAML that results, but I think that makes debugging harder and a worse UX. For example, in the UI when comparing different AppServers, I'd like to be able to compare just the 'configuration_extra_settings' values, to see how those were changed over time, and not have to guess what they were based on the final combined YAML (like we do now in the current version of the instance manager)